### PR TITLE
feat(tabs): add possibility to disable key navigations

### LINKED ISSUE #5537: Add possibility to disable keyNavigations on tabs.
**Is your feature request related to a problem? Please describe.**
I made edittable tab names by clicking in the tab heading, and there is an input tag to modify it. I solved it with custom template (tabHeading directive).
When i want to use arrows, shift+arrows or space, because of the keyNavigations, the input tag losing the focus.
I downgraded it to 5.0.0 version, and now works, but this is only a temporary solution.

**Describe the solution you'd like**
I want a variable to make possible to disable keyNavigations.

### LINKED ISSUE #5628: Allow disabling keyboard navigation for Tabsets
**Is your feature request related to a problem? Please describe.**

I am developing an application that has Tabs that work similarly to Excel Sheets. If you double click on the Tab name, it becomes an input field you can edit that modifies the name of the tab.

The issue is: since the Tabset has keyboard navigation, it conflicts with keyboard input on my element. I am unable to use space, home, arrows, etc.

**Describe the solution you'd like**

I would like a direct capable of disabling/enabling the keyboard navigation.

**Additional context**

The element:
image

Sample code:
```html
<tabset>
    <tab>
    <ng-template tabHeading>
        <span>Tab 1</span>
    </ng-template>
    Tab 1 content
    </tab>
    <tab>
    <ng-template tabHeading>
        <span><input type="text" autofocus value="Tab 2"/></span>
    </ng-template>
    Tab 2 content
    </tab>
</tabset>
```

### PR DESCRIPTION
**Pull request by Beni312**
possible closes #5628

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [ ] added/updated tests.
 - [x] added/updated API documentation.
 - [x] added/updated demos.

Fixed #5537

### CODE REVIEW

**Comment by daniloff200** [comment]
Hey, @Beni312 ! I see, that some units are broken in your PR. Could you please fix them?

**Comment by Beni312** [comment]
> Hey, @Beni312 ! I see, that some units are broken in your PR. Could you please fix them?
Hello daniloff200, 
I checked the error log, and it seems that the error doesn't link to my commit.
I runned the tests locally without error.
Could you run the tests again?

File: src/tabs/tabset.component.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<ul class="nav" [ngClass]="classMap" (click)="$event.preventDefault()">
+<ul *ngIf="isKeysAllowed" class="nav" [ngClass]="classMap" (click)="$event.preventDefault()">
```
Comment (valorkin) [inline src/tabs/tabset.component.html]: I believe disabling `keyNavActions` handler would be a better option than template duplication
performance wise too

File: src/tabs/tabset.component.ts [hunk 2]
```diff
@@ -40,6 +40,15 @@ export class TabsetComponent implements OnDestroy {
     this.setClassMap();
   }
 
+  /** provides possibility to set keyNavigations enable or disable, by default is enable */
+  @Input()
```
Comment (valorkin) [inline src/tabs/tabset.component.ts]: let's not bloat component API, and just use config

**Review by valorkin — DISMISSED** [review]
@Beni312 thank you for your contribution, please make this minor improvements and we are ready to go

**Comment by Beni312** [comment]
I removed the template duplicate and disable the key navigations in the function
did you mean this solution?

**Comment by daniloff200** [comment]
@Beni312  looks like something goes wrong, I see multiple issues with the build
Could you check them please?

**Comment by Beni312** [comment]
> @Beni312 looks like something goes wrong, I see multiple issues with the build
> Could you check them please?

@daniloff200 can you run the tests again?

**Comment by daniloff200** [comment]
@Beni312  that's not only tests
Please, check the Travis report : https://travis-ci.org/valor-software/ngx-bootstrap/builds/646459787?utm_medium=notification&utm_source=github_status

**Comment by Beni312** [comment]
@daniloff200, i fixed it, added a missing setter
Can you check it now?

**Comment by Beni312** [comment]
Hello @daniloff200, @valorkin 
Please review the code, i fixed the requests.
thanks

File: src/tabs/tabset.component.ts (hunk 2 above)
Comment (Beni312) [reply src/tabs/tabset.component.ts]: i changed to use the config, please review it

File: src/tabs/tabset.component.html (hunk 1 above)
Comment (Beni312) [reply src/tabs/tabset.component.html]: !

Comment (Beni312) [reply src/tabs/tabset.component.html]: !

**Review by daniloff200 — APPROVED** [review]

**Comment by daniloff200** [comment]
I've checked this functionality, looks like, everything works correctly

**Pull request merged**